nixos/modules/hardware/video/amdgpu-pro.nix

```
@@ -39,9 +39,14 @@ in
 
     hardware.firmware = [ package.fw ];
 
-    system.activationScripts.setup-amdgpu-pro = ''
-      ln -sfn ${package}/opt/amdgpu{,-pro} /run
-    '';
+    systemd.tmpfiles.settings.amdgpu-pro = {
+      "/run/amdgpu"."L+" =  {
+        argument = "${package}/opt/amdgpu";
+      };
+      "/run/amdgpu-pro"."L+" =  {
+        argument = "${package}/opt/amdgpu-pro";
+      };
```

```suggestion
      "/run/amdgpu"."L+".argument = "${package}/opt/amdgpu";
      "/run/amdgpu-pro"."L+".argument = "${package}/opt/amdgpu-pro";
```
I am a bit confused why the normal tmpfiles.d syntax is not used 🤔

---

This is a new way to specify tmpfiles rules, which allows overriding thanks to the use of attrsets instead of lists.